### ISSUE TITLE
option abortonclose only works in defaults section, not in backend section

### ISSUE THREAD

**Issue by robmueller** [haproxy/haproxy#2344]
### Detailed Description of the Problem

The `option abortonclose` appears to only work correctly when set in the `defaults` section. Setting `option abortonclose` on a single `backend` section does not appear to cause backend connections to be aborted  if any corresponding `frontend` request is aborted.

### Expected Behavior

Setting `option abortonclose` on a single `backend` should cause a connections to that `backend` to be aborted if a corresponding `frontend` connection is aborted.

### Steps to Reproduce the Behavior

I've created a trivial client and server to test.

```
cat > /tmp/c.pl <<EOF
use IO::Socket::IP;
use IO::Select;

my $c = IO::Socket::IP->new(PeerAddr => '127.0.0.1:8090');
print "C: connected to 127.0.0.1:8090\n";
print "C: sending request\n";
print $c <<EOF;
GET / HTTP/1.0
Host: foo.com

EOF

print "C: waiting 3 seconds\n";
sleep(3);
print "C: closed connection after 3 seconds\n";
EOF
```

```
cat > /tmp/s.pl <<EOF
use IO::Socket::IP;
use IO::Select;

my $l = IO::Socket::IP->new(LocalAddr => '127.0.0.1:8091', Listen => 5);
print "S: listening on 127.0.0.1:8091\n";
while (my $s = $l->accept) {
  print "S: new connection\n";
  while (<$s>) {
    print "S: $_";
    last if /^\r?\n$/;
  }
  print "S: read request complete, waiting\n";
  my $t = time;
  my $w = IO::Select->new();
  $w->add($s);
  $w->can_read();
  print "S: connection closed after " . (time - $t) . "s\n";
}
EOF
```

```
cat > /tmp/t.pl <<EOF
global
  log stdout daemon
  user nobody
  group nogroup
  nbthread 1

defaults
  log global
  mode http
  option httplog
  timeout client 20s
  timeout server 10s
  option abortonclose

frontend test
  bind 127.0.0.1:8090
  mode http
  option httpclose
  default_backend test

backend test
  server test1 127.0.0.1:8091
EOF
```

```
# perl /tmp/s.pl &
[1] 1116798
S: listening on 127.0.0.1:8091
# /usr/local/sbin/haproxy -d -f /tmp/t.cfg &
[2] 1118959
[NOTICE]   (1119410) : haproxy version is 2.8.3-86e043a
[NOTICE]   (1119410) : path to executable is /usr/local/sbin/haproxy
...
Using epoll() as the polling mechanism.

# perl /tmp/c.pl
C: connected to 127.0.0.1:8090
C: sending request
C: waiting 3 seconds
00000000:test.accept(0005)=0008 from [127.0.0.1:37150] ALPN=<none>
00000000:test.clireq[0008:ffffffff]: GET / HTTP/1.0
00000000:test.clihdr[0008:ffffffff]: host: foo.com
S: new connection
S: GET / HTTP/1.0
S: host: foo.com
S: connection: keep-alive
S: 
S: read request complete, waiting
C: closed connection after 3 seconds
00000000:test.srvcls[0008:0009]
00000000:test.clicls[0008:0009]
00000000:test.closed[0008:0009]
<30>Nov 12 06:28:05 haproxy[1118959]: 127.0.0.1:37150 [12/Nov/2023:06:28:02.718] test test/test1 0/0/0/-1/3000 400 206 - - CH-- 1/1/0/0/0 0/0 "GET / HTTP/1.0"
S: connection closed after 3s
```
In the above, we see our trivial client sending a request, and then closing the connection after 3 seconds. We see that the backend connection is immediately aborted when the client connection closes.

If we change to `option abortonclose` on just the backend.

```
cat > /tmp/t.pl <<EOF
global
  log stdout daemon
  user nobody
  group nogroup
  nbthread 1

defaults
  log global
  mode http
  option httplog
  timeout client 20s
  timeout server 10s

frontend test
  bind 127.0.0.1:8090
  mode http
  option httpclose
  default_backend test

backend test
  server test1 127.0.0.1:8091
  option abortonclose
EOF
```

```
# /usr/local/sbin/haproxy -d -f /tmp/t.cfg &
[2] 1119410
[NOTICE]   (1118959) : haproxy version is 2.8.3-86e043a
[NOTICE]   (1118959) : path to executable is /usr/local/sbin/haproxy
...
Using epoll() as the polling mechanism.

# perl /tmp/c.pl
C: connected to 127.0.0.1:8090
C: sending request
00000000:test.accept(0005)=0008 from [127.0.0.1:38994] ALPN=<none>
00000000:test.clireq[0008:ffffffff]: GET / HTTP/1.0
00000000:test.clihdr[0008:ffffffff]: host: foo.com
S: new connection
S: GET / HTTP/1.0
S: host: foo.com
S: connection: keep-alive
S: 
S: read request complete, waiting
C: waiting 3 seconds
C: closed connection after 3 seconds
#
00000000:test.srvcls[0008:0009]
00000000:test.clicls[0008:0009]
00000000:test.closed[0008:0009]
<30>Nov 12 06:37:27 haproxy[1119410]: 127.0.0.1:38994 [12/Nov/2023:06:37:17.427] test test/test1 0/0/0/-1/10002 504 198 - - sH-- 1/1/0/0/0 0/0 "GET / HTTP/1.0"
S: connection closed after 10s
```

In this case we can see the client cropped the connection after 3 seconds, but the backend connection was not aborted, and instead timed out after 10 seconds as configured via `timeout server 10s`

### What is your configuration?

```haproxy
global
  log stdout daemon
  user nobody
  group nogroup
  nbthread 1

defaults
  log global
  mode http
  option httplog
  timeout client 20s
  timeout server 10s
#1  option abortonclose

frontend test
  bind 127.0.0.1:8090
  mode http
  option httpclose
  default_backend test

backend test
  server test1 127.0.0.1:8091
#2  option abortonclose
```

### Output of `haproxy -vv`

```plain
HAProxy version 2.8.3-86e043a 2023/09/07 - https://haproxy.org/
Status: long-term supported branch - will stop receiving fixes around Q2 2028.
Known bugs: http://www.haproxy.org/bugs/bugs-2.8.3.html
Running on: Linux 5.10.0-24-amd64 #1 SMP Debian 5.10.179-5 (2023-08-08) x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = cc
  CFLAGS  = -O2 -g -Wall -Wextra -Wundef -Wdeclaration-after-statement -Wfatal-errors -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference -fwrapv -Wno-address-of-packed-member -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wno-cast-function-type -Wno-string-plus-int -Wno-atomic-alignment
  OPTIONS = USE_OPENSSL=1 USE_ZLIB=1 USE_PCRE=1
  DEBUG   = -DDEBUG_STRICT -DDEBUG_MEMORY_POOLS

Feature list : -51DEGREES +ACCEPT4 +BACKTRACE -CLOSEFROM +CPU_AFFINITY +CRYPT_H -DEVICEATLAS +DL -ENGINE +EPOLL -EVPORTS +GETADDRINFO -KQUEUE -LIBATOMIC +LIBCRYPT +LINUX_CAP +LINUX_SPLICE +LINUX_TPROXY -LUA -MATH -MEMORY_PROFILING +NETFILTER +NS -OBSOLETE_LINKER +OPENSSL -OPENSSL_WOLFSSL -OT +PCRE -PCRE2 -PCRE2_JIT -PCRE_JIT +POLL +PRCTL -PROCCTL -PROMEX -PTHREAD_EMULATION -QUIC +RT +SHM_OPEN -SLZ +SSL -STATIC_PCRE -STATIC_PCRE2 -SYSTEMD +TFO +THREAD +THREAD_DUMP +TPROXY -WURFL +ZLIB

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_TGROUPS=16, MAX_THREADS=256, default=4).
Built with OpenSSL version : OpenSSL 1.1.1w  11 Sep 2023
Running on OpenSSL version : OpenSSL 1.1.1w  11 Sep 2023
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built with network namespace support.
Built with zlib version : 1.2.11
Running on zlib version : 1.2.11
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE version : 8.39 2016-06-14
Running on PCRE version : 8.39 2016-06-14
PCRE library supports JIT : no (USE_PCRE_JIT not set)
Encrypted password support via crypt(3): yes
Built with gcc compiler version 10.2.1 20210110

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
         h2 : mode=HTTP  side=FE|BE  mux=H2    flags=HTX|HOL_RISK|NO_UPG
       fcgi : mode=HTTP  side=BE     mux=FCGI  flags=HTX|HOL_RISK|NO_UPG
  <default> : mode=HTTP  side=FE|BE  mux=H1    flags=HTX
         h1 : mode=HTTP  side=FE|BE  mux=H1    flags=HTX|NO_UPG
  <default> : mode=TCP   side=FE|BE  mux=PASS  flags=
       none : mode=TCP   side=FE|BE  mux=PASS  flags=NO_UPG

Available services : none

Available filters :
	[BWLIM] bwlim-in
	[BWLIM] bwlim-out
	[CACHE] cache
	[COMP] compression
	[FCGI] fcgi-app
	[SPOE] spoe
	[TRACE] trace
```

**Comment by capflam** [comment]
Indeed, I can confirm the issue. I'm on a fix.

**Comment by capflam** [comment]
I pushed a fix in 2.9-DEV. I will backport it and it will be shipped with the 2.8.4. Thanks !
